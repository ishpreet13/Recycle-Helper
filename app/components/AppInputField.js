import React from "react";
import { StyleSheet, View, TouchableOpacity } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import colors from "../config/colors";
import { TextInput } from "react-native-gesture-handler";

function AppInputField({
  leftIcon,
  rightIcon,
  placeholder,
  handlePasswordVisibility,
  ...rest
}) {
  return (
    <View style={styles.container}>
      {leftIcon ? (
        <MaterialCommunityIcons
          name={leftIcon}
          size={20}
          style={styles.leftIcon}
        />
      ) : null}
      <TextInput
        {...rest}
        placeholder={placeholder}
        placeholderTextColor={colors.placeHolderTextColor}
        style={styles.input}
      ></TextInput>
      {rightIcon ? (
        <TouchableOpacity onPress={handlePasswordVisibility}>
          <MaterialCommunityIcons
            name={rightIcon}
            size={20}
            style={styles.rightIcon}
          />
        </TouchableOpacity>
      ) : null}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.white,
    borderRadius: 20,
    borderStyle: "solid",
    borderColor: colors.black,
    borderWidth: 1,
    flexDirection: "row",
    padding: 12,
    width: "100%",
    alignItems: "center",
    marginBottom: 10,
  },
  input: {
    flex: 1,
    width: "100%",
    fontSize: 18,
  },
  leftIcon: {
    marginRight: 10,
    color: colors.black,
  },
  rightIcon: {
    marginLeft: 10,
    alignSelf: "center",
  },
});

export default AppInputField;
