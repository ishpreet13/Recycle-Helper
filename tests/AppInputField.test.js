import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import AppInputField from "../app/components/AppInputField";

describe("Testing AppInputField Component ", () => {
  const wrapper = renderer.create(<AppInputField />);
  it("should render", () => {
    expect(wrapper.toJSON()).toBeTruthy();
  });
});

describe("<AppInputField />", () => {
  let tree;
  it("should render correctly", async () => {
    tree = renderer.create(<AppInputField />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
