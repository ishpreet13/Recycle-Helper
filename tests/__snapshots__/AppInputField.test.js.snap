// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppInputField /> should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#000",
      "borderRadius": 20,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "marginBottom": 10,
      "padding": 12,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    placeholderTextColor="#a9a9a9"
    rejectResponderTermination={true}
    style={
      Object {
        "flex": 1,
        "fontSize": 18,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
